### ISSUE TITLE
Handle country codes for countries related to the united states with area code part of the country calling code +1-xxx

### ISSUE THREAD

**Issue by Startouf** [floere/phony#479]
What I did + What happened
----------

We have a backend service that validates both the country code and the rest of the phone number, we took advantage of Phony to validate not like this

```
Phony.plausible?('+18687620266')
=> true
```
﻿
but like this : 

```
﻿Phony.plausible?('18687620266', cc: '1868')
=> false
```

And the current behavior is that only the following works :

```
Phony.plausible?('18687620266', cc: '1')
=> true
```

What I expected to happen and why
---------------------------------

Many phone frontend libraries have implemented country codes for countries that share the united states indicator +1 but with an additional region code that becomes part of the country code 

Take for instance [Trinidad and Tobago](https://en.wikipedia.org/wiki/Trinidad_and_Tobago)

https://countrycode.org/trinidadandtobago
" +1-868 COUNTRY CALLING CODE"

Such country codes are often exposed by libraries 
- https://baseweb.design/components/phone-input
﻿- https://bl00mber.github.io/react-phone-input-2.html
﻿- https://kaihotz.github.io/React-PhoneNr-Input/
﻿
![Screenshot 2021-03-02 at 20 18 11](https://user-images.githubusercontent.com/7388889/109702368-6da77780-7b94-11eb-9319-1c83abc7c9f3.png)

So it would be legitimate to handle such country codes, and we could expect the following to work

```
Phony.plausible?('18687620266', cc: '1868')
=> true
```

Can you confirm such a behavior would be desirable ?

**Comment by floere** [comment]
@Startouf Thanks first of all, Cyril, for this excellent issue entry!

I am in two minds about this – there's this longstanding "issue" of whether Phony should include country handling or whether it should be a pure E164 library (the original goal).
Also, from the E164 view, we get a weird naming, since the CC does not uniquely determine a "country", and the NDC is therefore not necessarily a "national" destination code. And that results in problems with naming in the API. Since it's a E164 library, `cc` only refers to the cc, but has only by initial intent anything to do with the country, and is per convention 1-3 digits long.

So, currently, the above check in Phony would be done like so: `ruby -e 'require "phony"; p Phony.plausible?("18687620266", cc: "1", ndc: "868")' # => true`. CC and NDC sound weird, but adhere to E164.

Splitting up a `cc: "1868"` -> `cc: "1", ndc: "868"` internally might be an option, but then the expectation may come up that a CC being 4 digits long is accepted everywhere in Phony.

Essentially, I can't confirm at this point in time.

**Comment by floere** [comment]
Note to self: A simple solution might be `Phony.plausible?("18687620266", ccc: "1868")`, which then simply splits the `ccc` into `cc` and `ndc`. But how to define it? E.g. `7x` is RU + KZ, with `76` and `77` being KZ.

**Comment by floere** [comment]
I've released a basic version of this in 2.20.0, see https://github.com/floere/phony/blob/master/qed/plausibility.md#ccc-country-calling-code.
